cupy/_core/_routines_sorting.pyx

```diff
@@ -231,21 +231,67 @@ cdef _ndarray_base _ndarray_argpartition(self, kth, axis):
     ndim = data._shape.size()
     _axis = internal._normalize_axis_index(_axis, ndim)
 
+    if axis == ndim - 1:
+        data = self
+    else:
+        data = _manipulation.rollaxis(self, axis, ndim).copy()
+
     length = data._shape[_axis]
```

Comment:
Shoulde be prior to rollaxis?

Reply:
It seems that partition does this as well: https://github.com/cupy/cupy/blob/9d439227f1a53bbd9ebb16784e40bcce44bc8042/cupy/_core/_routines_sorting.pyx#L145

Reply:
ah, this line should be:
```suggestion
    length = self._shape[_axis]
```
as the axis along which to sort is rolled at the last in `data` when `axis != ndim - 1`